crates/router/src/services.rs

```diff
@@ -45,4 +45,29 @@ impl Store {
         // Example: {shard_5}_drainer_stream
         format!("{{{}}}_{}", shard_key, self.config.drainer_stream_name,)
     }
+
+    #[cfg(feature = "kv_store")]
+    pub(crate) async fn push_to_drainer_stream<T>(
+        &self,
+        redis_entry: storage_models::kv::TypedSql,
+        partition_key: crate::utils::storage_partitioning::PartitionKey<'_>,
+    ) -> crate::core::errors::CustomResult<(), crate::core::errors::StorageError>
+    where
+        T: crate::utils::storage_partitioning::KvStorePartition,
+    {
+        use error_stack::ResultExt;
+
+        let shard_key = T::shard_key(partition_key, self.config.drainer_num_partitions);
```

Comment:
the second argument `self.config.drainer_num_partitions` can be moved into `shard_key` function itself?.